messaging/src/test/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueueTest.java

```diff
@@ -0,0 +1,211 @@
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.IdentifierFactory;
+import org.axonframework.common.jpa.EntityManagerProvider;
+import org.axonframework.common.jpa.SimpleEntityManagerProvider;
+import org.axonframework.common.transaction.Transaction;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.deadletter.DeadLetter;
+import org.axonframework.messaging.deadletter.GenericDeadLetter;
+import org.axonframework.messaging.deadletter.SequencedDeadLetterQueue;
+import org.axonframework.messaging.deadletter.SequencedDeadLetterQueueTest;
+import org.junit.jupiter.api.*;
+
+import java.sql.SQLException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.stream.IntStream;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class JpaSequencedDeadLetterQueueTest extends SequencedDeadLetterQueueTest<EventMessage<?>> {
+
+    private final TransactionManager transactionManager = spy(new NoOpTransactionManager());
+    EntityManagerFactory emf = Persistence.createEntityManagerFactory("dlq");
+    EntityManager entityManager = emf.createEntityManager();
+    private EntityTransaction transaction;
+
+    @BeforeEach
+    public void setUpJpa() throws SQLException {
+        transaction = entityManager.getTransaction();
+        transaction.begin();
+    }
+
+    @AfterEach
+    public void rollback() {
+        transaction.rollback();
+    }
+
+    @Override
+    protected void setClock(Clock clock) {
+        GenericDeadLetter.clock = clock;
+    }
+
+    @Override
+    public DeadLetter<EventMessage<?>> generateInitialLetter() {
+        return new GenericDeadLetter<>("sequenceIdentifier", generateEvent(), generateThrowable());
+    }
+
+    @Override
+    protected DeadLetter<EventMessage<?>> generateFollowUpLetter() {
+        return new GenericDeadLetter<>("sequenceIdentifier", generateEvent());
+    }
+
+    @Override
+    protected DeadLetter<EventMessage<?>> mapToQueueImplementation(DeadLetter<EventMessage<?>> deadLetter) {
+        if (deadLetter instanceof JpaDeadLetter) {
+            return deadLetter;
+        }
+        if (deadLetter instanceof GenericDeadLetter) {
+            return new JpaDeadLetter<>(IdentifierFactory.getInstance().generateIdentifier(),
+                                       0L,
+                                       ((GenericDeadLetter<EventMessage<?>>) deadLetter).getSequenceIdentifier()
+                                                                                        .toString(),
+                                       GenericDeadLetter.clock.instant(),
```

Comment:
Why change the time when mapping from the given `DeadLetter` to the queue specific implementation?

Reply:
Fair point, changed it